Sources/GRPC/Compression/Zlib.swift

```diff
@@ -69,12 +67,15 @@ enum Zlib {
           self.stream.availableInputBytes = 0
         }
         
-        return try output.writeWithUnsafeMutableBytes(minimumWritableBytes: Int(upperBound)) { outputPointer in
+        let writtenBytes = try output.writeWithUnsafeMutableBytes(minimumWritableBytes: Int(upperBound)) { outputPointer in
           try self.stream.deflate(
              outputBuffer: CGRPCZlib_castVoidToBytefPointer(outputPointer.baseAddress!),
              outputBufferSize: outputPointer.count
           )
         }
+        
+        let bytesRead = inputPointer.count - self.stream.availableInputBytes
+        return (bytesRead, writtenBytes)
```

Comment:
@glbrntt I think this should fix the issue with the pointer being accessed while we already moved out of the function